### PR TITLE
AC Automated Test - LRAC 8835 | master

### PR DESCRIPTION
**Pull request by AfonsoHSantos**
https://issues.liferay.com/browse/LRAC-8835

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro [hunk 1]
```diff
@@ -1,5 +1,27 @@
 definition {
 
+	macro assertDXPContactCountPreview {
+		key_numberContacts = "${numberContacts}";
+
+		ACDXPSettings.goToInstanceSettingsAC();
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro:6]: @AfonsoHSantos, I think you can put this navigation in your test

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+
+		Click(locator1 = "ACDXPSettings#TOKEN_INPUT_FIELD");
+
+		Paste(locator1 = "ACDXPSettings#TOKEN_INPUT_FIELD");
+
+		Click(locator1 = "ACDXPSettings#CONNECT_DISCONNECT_BUTTON");
+
+		ACDXPSettings.assertSuccessfulConnection();
+
+		ACDXPSettings.navToSyncedContactDataMenu();
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro:16]: @AfonsoHSantos, I think you can put this navigation in your test

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+
+		ACDXPSettings.goToInstanceSettingsAC();
+
+		Click(locator1 = "ACDXPSettings#TOKEN_INPUT_FIELD");
+
+		Paste(locator1 = "ACDXPSettings#TOKEN_INPUT_FIELD");
+
+		Click(locator1 = "ACDXPSettings#CONNECT_DISCONNECT_BUTTON");
+
+		ACDXPSettings.assertSuccessfulConnection();
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro:14]: Hi @AfonsoHSantos, I think you can use `ACDXPSettings.connectDXPtoAnalyticsCloud();` and after you go to sync contact and do the assert

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro [hunk 4]
```diff
@@ -1,5 +1,27 @@
...
+
+		ACDXPSettings.navToSyncedContactDataMenu();
+
+		Click(locator1 = "ACDXPSettings#SYNC_CONTACTS_OPTION");
+
+		AssertElementPresent(
+			key_numberContacts = "${numberContacts}",
+			locator1 = "ACDXPSettings#SYNC_ALL_CONTACTS_NUMBER");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDXPSettings.macro:22]: Hi @AfonsoHSantos, I think you can go to sync contact make a assert in the number and after you create a new user and return to sync contact and make another assert in the number

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AnalyticsCloudConnectionTest.testcase [hunk 5]
```diff
@@ -74,12 +74,11 @@ definition {
 	}
 
 	@description = "Automation ID: LRAC-8835 | Test Summary: Assert Liferay Data Source Contact Count Preview"
-	@ignore = "true"
 	@priority = "4"
 	test AssertLiferayDataSourceContactCountPreview {
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AnalyticsCloudConnectionTest.testcase:78]: **Steps to this test:**

- Connect DXP to Analytics Cloud with sync by organizations
- Go to sync contact > Check the number
- Create a user
- Go to sync contact > Make a sync contact again
- Check the number

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AnalyticsCloudConnectionTest.testcase [hunk 6]
```diff
@@ -74,12 +74,11 @@ definition {
...
-
-		// TODO LRAC-8835 AssertLiferayDataSourceContactCountPreview
-
+		task ("Go to AC/DXP Settings and assert the number of contacts") {
+			ACDXPSettings.assertDXPContactCountPreview(numberContacts = "1");
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AnalyticsCloudConnectionTest.testcase:80]: hi @AfonsoHSantos , keep in mind to create macros more generic and reusable with some defined purpose (actions, navigation, asserts, etc), the macros should be like a library/method with actions that we can reuse in other tests, this macro is too restricted to the context of your test only.

**Review by alexcandidos — CHANGES_REQUESTED** [review]
hi @AfonsoHSantos , could you please review the suggestions and send a new PR? 

Thanks!

**Comment by AfonsoHSantos** [comment]
hi @alexcandidos @ViniLopes20 , resent on #845

**Pull request closed without merging**